release/models/aft/openconfig-aft-common.yang

```diff
@@ -244,11 +258,134 @@ submodule openconfig-aft-common {
           }
         }
 
+        container encap-headers {
+          description
+            "Container for packet encapsulation headers.  When present, this
+            container indicates that encapsulation of the packet matching the
+            next-hop is performed using a stack of one or more packets defined
+            in the list encap-header.
+
+            Each entry in the list must indicate an encapsulation type and
+            populate a container with the parameters for that encapsulation
+            header.";
+          list encap-header {
+            description
+              "A list of headers added on top of a packet.  The first entry
+              in the list is the inner-most packet at the bottom of the
+              stack.
+
+              For example, in an encapsulation stack with MPLS in UDP, the
+              first index in the list is the MPLS packet and the second
+              index is UDP.";
+            key "index";
+
+            leaf index {
+              type leafref {
+                path "../state/index";
+              }
+              description
+                "A unique index identifying an encapsulation header in a stack
+                of encapsulation headers.";
+            }
+
+            container state {
+              description
+                "State parameters relating to encapsulation headers.";
+              uses aft-common-nexthop-encap-headers-state;
+            }
+
+            container gre {
+              when "../state/type = 'oc-aftt:GRE'";
+              description
+                "Container of nodes for GRE encapsulation.";
+              container state {
+                description
+                  "State parameters relating to GRE encapsulation headers.";
+                uses aft-common-entry-nexthop-gre-state;
+              }
+            }
+
+            container ipv4 {
```

Comment:
Should we have IPv4 and IPv6 separately here? It means there is a change from the existing model where one could have an `ip-in-ip` encap that would be either `ip-in-ipv4` or `ip-in-ipv6` based on the src/dst addresses that were sent. Are we OK with having to have explicitly different leaves here?

Reply:
Note that having this as separate per-AFI `container` conflicts with the approach used for UDP below, where UDP is implying UDPv4 or UDPv6 and the `src-addr` leaf can be either AFI. Is there a reason not to have IPv[46] combined when UDPoIPv[46] is?